src/lib/api/commonUtils.ts

```diff
@@ -9,6 +9,7 @@ interface Options {
   method: HttpMethod;
   headers: Record<string, string>;
   body?: string;
+  cache?: RequestCache;
```

Comment:
I suggest that the `fetch()` function should accept any `options` as its second argument. This is because there are many scenarios where options other than `cache` may be needed, such as revalidation using `next: { revalidate: 3600 }`. This suggestion is based on the OCP [(Open-Closed Principle)](https://www.freecodecamp.org/news/open-closed-principle-solid-architecture-concept-explained/#:~:text=The%20open%2Dclosed%20principle%20encourages,following%20the%20open%2Dclosed%20principle.) of the SOLID principles.

You can find the complete list of available options documented here: https://developer.mozilla.org/en-US/docs/Web/API/fetch.

Or, do you have any reasons to limit the option?

Reply:
Thank you for your suggestion and our discussion!
I fixed this part with based on OCP [(Open-Closed Principle)](https://www.freecodecamp.org/news/open-closed-principle-solid-architecture-concept-explained/#:~:text=The%20open%2Dclosed%20principle%20encourages,following%20the%20open%2Dclosed%20principle.) of the SOLID principles as you mentioned. so please take a look!

Reply:
Thank you! It looks good!